themes/rjorgenson/rjorgenson.theme.sh

```diff
@@ -14,88 +14,82 @@ SCM_THEME_PROMPT_CLEAN=" ${bold_green}✓${normal}"
 SCM_GIT_CHAR="${STRING_COLOR}±${normal}"
 SCM_SVN_CHAR="${bold_cyan}⑆${normal}"
 SCM_HG_CHAR="${bold_red}☿${normal}"
+PROMP_CHAR="${BRACKET}➞ ${normal}"
 
 #Mysql Prompt
 export MYSQL_PS1="(\u@\h) [\d]> "
 
+TITLEBAR=""
 case $TERM in
-        xterm*)
-        TITLEBAR="\[\033]0;\w\007\]"
-        ;;
-        *)
-        TITLEBAR=""
-        ;;
+        xterm*) TITLEBAR="\[\033]0;\w\007\]" ;;
 esac
 
-PS3=">> "
+if [[ $OSTYPE =~ "darwin" ]]; then
+    echo "${BRACKET}➞  ${normal}"
+fi
 
 __my_rvm_ruby_version() {
-    local gemset=$(echo $GEM_HOME | awk -F'@' '{print $2}')
-  [ "$gemset" != "" ] && gemset="@$gemset"
-    local version=$(echo $MY_RUBY_HOME | awk -F'-' '{print $2}')
-    local full="$version$gemset"
-  [ "$full" != "" ] && echo "${BRACKET_COLOR}[${STRING_COLOR}$full${BRACKET_COLOR}]${normal}"
+    local gemset ; gemset=$(echo "${GEM_HOME}" | awk -F'@' '{print $2}')
+    local version ; version=$(echo "${MY_RUBY_HOME}" | awk -F'-' '{print $2}')
+    [ "${gemset}" != "" ] && gemset="@${gemset}"
+    local full="${version}${gemset}"
+    [ "${full}" != "" ] && echo "${BRACKET}[${STRING_COLOR}${full}${BRACKET}]${normal}"
 }
 
 is_vim_shell() {
-        if [ ! -z "$VIMRUNTIME" ]
-        then
-                echo "${BRACKET_COLOR}[${STRING_COLOR}vim shell${BRACKET_COLOR}]${normal}"
-        fi
+    if [ ! -z "$VIMRUNTIME" ]
+    then
+        echo "${BRACKET}[${STRING_COLOR}vim shell${BRACKET}]${normal}"
+    fi
 }
 
 function is_integer() { # helper function for todo-txt-count
     [ "$1" -eq "$1" ] > /dev/null 2>&1
         return $?
 }
 
+# XXX do we need/want to integrate with todo.sh? We don't provide it and I
+# can't find a version online that accepts ls as an input
 todo_txt_count() {
     if _omb_util_command_exists todo.sh; then # is todo.sh installed
-        count=$(todo.sh ls | egrep "TODO: [0-9]+ of ([0-9]+) tasks shown" | awk '{ print $4 }')
-        if is_integer $count; then # did we get a sane answer back
-            echo "${BRACKET_COLOR}[${STRING_COLOR}T:$count${BRACKET_COLOR}]$normal"
+        count=$(todo.sh ls \
+          | awk '/TODO: [0-9]+ of ([0-9]+) tasks shown/ { print $4 }')
+        if [ "${count}" -eq "${count}" ] &> /dev/null
+        then # did we get a sane answer back
+            echo "${BRACKET}[${STRING_COLOR}T:$count${BRACKET}]$normal"
         fi
     fi
 }
 
 modern_scm_prompt() {
-        CHAR=$(scm_char)
-        if [ $CHAR = $SCM_NONE_CHAR ]
-        then
-                return
-        else
-                echo "${BRACKET_COLOR}[${CHAR}${BRACKET_COLOR}][${STRING_COLOR}$(scm_prompt_info)${BRACKET_COLOR}]$normal"
-        fi
-}
-
-my_prompt_char() {
-    if [[ $OSTYPE =~ "darwin" ]]; then
-        echo "${BRACKET_COLOR}➞  ${normal}"
-    else
-        echo "${BRACKET_COLOR}➞ ${normal}"
-    fi
+    CHAR=$(scm_char)
+    [ "${CHAR}" = "${SCM_NONE_CHAR}" ] \
+        || echo "${BRACKET}[${CHAR}${BRACKET}][${STRING_COLOR}$(scm_prompt_info)${BRACKET}]$normal"
```

Comment:
```suggestion
    [ "${CHAR}" = "${SCM_NONE_CHAR}" ] ||
        echo "${BRACKET}[${CHAR}${BRACKET}][${STRING_COLOR}$(scm_prompt_info)${BRACKET}]$normal"
```

This might depend on the coding style, but we can place `||` at the end of the line so that Bash knows that the line is continued without adding an extra backslash.

Reply:
I did not know BASH could do that.

I still prefer the the explicit line continue. It clearly indicates that the line starting with `||` or `&&` is a continuation of the test.

Reply:
I actually wanted to make the style consistent across the entire codebase, but it seems to be not the case at the moment (I searched them and found 36 cases of `command1 Op <newline> command2`, 18 cases of `command1 Op \<newline> command2`, and 4 cases of `command1 \<newline> Op command2`, where Op is `||` or `&&`). So that's fine if this is your preference. Nevertheless, I don't think the other patterns are less clear than this style. The line continuation is visibly indicated by the indentation after all.

Reply:
Anyway, let's keep the current style. I wouldn't request changing the style.